pkg/odo/cli/logs/logs.go

```diff
@@ -36,9 +34,10 @@ type LogsOptions struct {
 	clientset *clientset.Clientset
 
 	// variables
-	componentName string
-	contextDir    string
-	out           io.Writer
+	componentName    string
+	contextDir       string
+	out              io.Writer
+	containerLogChan chan map[string]interface{}
```

Comment:
It seems this variable can be defined as a local varibale in the Run method

Reply:
Yeah, can do. I don't have a preference for either.

Reply:
Varibales should always have the smallest possible scope. Having it in the Receiver makes it accessible from all the methods of the receiver, when it is only needed in the Run method.

Reply:
Variable removed in favour of a struct of channels being returned by business layer.